### ISSUE TITLE
Chartist CSS to Stylus

### ISSUE THREAD

**Issue by mcdonnelldean** [vidi-insights/vidi-dashboard#119]
What are your thoughts on this @thefoxis ?

My understanding is the correct way to style the charts is to do what we did, drop the default and use our own CSS. Given this is done should we now move it to Stylus for the benefits that gives?

If yes, go ahead and label as a change and do.

**Comment by thefoxis** [comment]
we'd want to use most of their CSS as it handles basic chart styling, so for customisation we'd want to alter the colours mostly (or maybe type if it's present). that's already done. the problem with porting to Stylus is that the Sass syntax uses lists (the [nth function](http://hugogiraudel.com/2013/07/15/understanding-sass-lists/)) and there seems to be no direct way of mapping that behaviour back to Stylus (or maybe it would be possible with [iteration](http://stylus-lang.com/docs/iteration.html)). Stylus is more limited so it might not be, would have to investigate more, so far I used [Sass to Stylus](http://sass2stylus.com/) and it failed on that statement too.

**Comment by mcdonnelldean** [comment]
Hmmm, so two questions come from this,

- Given we are going to lean on chartist heavily how much of their base CSS do we need? Just styles?

- Are we going to hit a point where we are blocked by stylus due to needing Sass to work with charts?

If we realistically need Sass to get the best out of the charts I'd be happy set the item of work as move from Stylus to Sass. Thoughts?

**Comment by thefoxis** [comment]
their entire styles can be found [here](https://github.com/gionkunz/chartist-js/tree/develop/src/styles). I can't see us changing the main file—`chartist.scss` as it includes displaying graphs, SVG styling and some math. what we **want** to modify are only [settings.scss](https://github.com/gionkunz/chartist-js/blob/develop/src/styles/settings/_chartist-settings.scss). so answering your question—I'd say we can port everything from Stylus to Sass to have variables in settings instead of pure CSS for easier colour theming but otherwise I don't see a reason to, unless you want to really alter the layout and display of the charts.

**Comment by mcdonnelldean** [comment]
@thefoxis Nope, if that's all that's needed I'm fine with that!
